source/cursorManager.py

```diff
@@ -59,8 +62,33 @@ def __init__(self, parent, cursorManager, text, caseSensitivity):
 		self.CentreOnScreen()
 		self.findTextField.SetFocus()
 
+	def updateSearchEntries(self, searchEntries, currentSearchTerm):
+		if not currentSearchTerm:
+			return
+		if not searchEntries:
+			searchEntries.insert(0, currentSearchTerm)
+			return
+		# we can not accept entries that differ only on text case because of a wxComboBox limitation on MS Windows
+		# see https://wxpython.org/Phoenix/docs/html/wx.ComboBox.html
+		#notice also that python 2 does not offer caseFold functionality so lower is the best we can have for comparing strings
+		for index, item in enumerate(searchEntries):
+			if(item.lower() == currentSearchTerm.lower()):
+				#if the user has selected a previous search term in the list or retyped an already listed term , we need to make sure the current search term becomes the first item of the list,
+				# so that it will appear selected by default when the dialog is shown again
+				# If the current search term differs from the current item only in case letters, we will choose to store the new search as we can not store both.
+				searchEntries.pop(index)
+				searchEntries.insert(0, currentSearchTerm)
+				return
+		# not yet listed. Save it.
+		if len(searchEntries) >= 20:

```

Comment:
Also use a constant for 19 say `LAST_TERM_INDEX`, then you can use it here. It also makes it easier to adjust the number of terms in the future.
```suggestion
		if len(searchEntries) > LAST_TERM_INDEX:
```